hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterFileSystem.java

```diff
@@ -382,4 +392,106 @@ public void stop() {
+      status = false;
+    }
+    return status;
+  }
+
+  // This API should be called on the 'new active cluster' to complete the process of switch over,
+  // to create the new active_cluster.id file.
+  public boolean performSwitchOver() {
+    LOG.info("Creating new cluster ID file during switchover");
+    boolean status = true;
+    int wait = conf.getInt(HConstants.THREAD_WAKE_FREQUENCY, 10 * 1000);
+    try {
+      String ms = conf.get(HConstants.HBASE_META_TABLE_SUFFIX,
+        HConstants.HBASE_META_TABLE_SUFFIX_DEFAULT_VALUE);
+      if (
+        !FSUtils.checkFileExistsInHbaseRootDir(fs, rootdir,
+          HConstants.ACTIVE_CLUSTER_SUFFIX_FILE_NAME, wait)
+      ) {
+        FSUtils.setActiveClusterSuffix(fs, rootdir, new ActiveClusterSuffix(ms), wait);
+      }
+      activeClusterSuffix = FSUtils.getActiveClusterSuffix(fs, rootdir);
+    } catch (IOException ioe) {
+      // reset the cluster id
+      status = false;
+      LOG.error("Failed to create active cluster id file during switchover, exception", ioe);
+    }
+    return status;
+  }
+
+  private void negotiateActiveClusterIdFile(long wait) throws IOException {
+    LOG.info("Active cluster Suffix file");
+    String suffix_file_name = HConstants.ACTIVE_CLUSTER_SUFFIX_FILE_NAME;
+    boolean activeClusterSuffixFileExists =
+      FSUtils.checkFileExistsInHbaseRootDir(fs, rootdir, suffix_file_name, wait);
+
+    if (!isReadOnlyModeEnabled(conf)) {
+      // this is the active cluster, create active cluster suffix file if it does not exist
+      if (!activeClusterSuffixFileExists) {
+        FSUtils.setActiveClusterSuffix(fs, rootdir,
+          new ActiveClusterSuffix(getActiveClusterSuffixFromConfig(conf)), wait);
+      } else {
+        // verify the contents against the config set
+        ActiveClusterSuffix acs = FSUtils.getActiveClusterSuffix(fs, rootdir);
+        if (
+          acs.equals(getActiveClusterSuffix())
+            || (acs.toString().equals("null") && getActiveClusterSuffix().toString().isEmpty())
+        ) {
+          LOG.info("Active cluster is being restarted, Suffix {}", acs);
+        } else {
+          // throw error
+          LOG.info("rootdir {} : Active Cluster File Suffix {} ", rootdir, acs);
+          throw new IOException("Can not start active cluster, please check the value of "
+            + "configuration hbase.global.readonly.enabled in hbase-site.xml, "
+            + "Active Cluster Suffix : " + getActiveClusterSuffix()
+            + ": Active Cluster File Suffix " + acs);
+        }
+      }
+    } else {
+      // this is a replica cluster
+      if (!activeClusterSuffixFileExists) {
+        // suffix file must exist, if not throw error
+        LOG.info("rootdir {}", rootdir);
+        throw new IOException("Read Replica Cluster must have associated Active Cluster Suffix "
+          + "file. File 'active_cluster_suffix.id' does not exist in $HBASE_ROOT.");
+      }
```

Comment:
This case doesn't necesserily have to be an error: users should be able to start the read replica clusters without and active cluster running. The new file is actually only important for non-RO mode clusters, RO instances don't need to check it at all.